query/math.go

```diff
@@ -46,6 +46,12 @@ func (v MathValuer) Call(name string, args []interface{}) (interface{}, bool) {
 			return v.callTrigFunction(math.Cos, arg0)
 		case "tan":
 			return v.callTrigFunction(math.Tan, arg0)
+		case "floor":
```

Comment:
For these functions, it should be easier to just do this:

```
switch arg0 := arg0.(type) {
case float64:
    return math.Floor(arg0), true
case int64, uint64:
    return arg0, true
default:
    return nil, true
}
```

I wouldn't worry about doing type checking here. While it's possible for us to call the function incorrectly in testing, the type checker should take care of things during actual execution.

I don't think you need a separate function and indirection for this since it's rather simple.